### PR TITLE
[BEAM-4425] Force load JDBC drivers

### PR DESCRIPTION
**Pull request by kennknowles**
In some environments (known to be OSX) the JDBC drivers are not loaded by the time they are used. This is a cherry-pick from #5490 to make it easy for anyone to review just these commits.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

### CODE REVIEW

**Comment by kennknowles** [comment]
@dkulp @jbonofre @timrobertson100 @boyuanzz

File: sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/impl/JdbcDriverTest.java [hunk 1]
```diff
@@ -32,11 +32,17 @@
...
 import org.apache.calcite.jdbc.CalciteConnection;
+import org.junit.Before;
 import org.junit.Test;
 
 /** Test for {@link JdbcDriver}. */
 public class JdbcDriverTest {
 
+  @Before
+  public void before() throws Exception {
+    Class.forName("org.apache.beam.sdk.extensions.sql.impl.JdbcDriver");
```
Comment (apilloud) [inline sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/impl/JdbcDriverTest.java]: I'd like to understand the environments that require this. What runtime do they have? This should be handled by the service file for java.sql.Driver in Java 7+. How do we test that path keeps working?

Comment (kennknowles) [reply sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/impl/JdbcDriverTest.java]: If it varies on env, I think we'd need a test matrix. We used to have a bit but shut down our Travis job.

Comment (kennknowles) [reply sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/impl/JdbcDriverTest.java]: TODO(me): when I am at the computers where it is broken, report my env to the JIRA.

**Comment by timrobertson100** [comment]
I have not investigated at all, other than to apply the changes and run the tests.

I can confirm tests now pass on my environment (macbook pro, high Sierra 10.13.4, Java 1.8.0_172-b11).

Without them I saw:
  java.sql.SQLException: No suitable driver found for jdbc:calcite
  java.sql.SQLException: No suitable driver found for jdbc:beam

**Comment by apilloud** [comment]
The tests pass without this change on Linux openjdk 1.8.0_131, OS X Sierra 10.12.6 with Java 1.8.0_141-b15, and OS X Sierra 10.12.6 upgraded to Java 1.8.0_172-b11.

**Comment by timrobertson100** [comment]
Interesting @apilloud  - how are you launching the tests please to ensure we are equivalent?

**Comment by apilloud** [comment]
I think that might be it. I am running this:  `./gradlew :beam-sdks-java-extensions-sql:test`

**Comment by apilloud** [comment]
LGTM.

The thing this is suppose to be testing isn't working, but the tests pass anyway because a bunch of other things also load the JDBC driver. I can repro this if I ensure the JdbcDriverTest is the first thing run. That makes it an ordering issue.

**Comment by vectorijk** [comment]
w/o loading `Class.forName("")`, it failed on my env. paste my env: command `./gradlew :beam-sdks-java-extensions-sql:test` macOS, java 1.8.0_112

**Comment by kennknowles** [comment]
OK, merging to fix build and we should track in JIRA that the service loader pattern isn't working.

**Pull request merged**